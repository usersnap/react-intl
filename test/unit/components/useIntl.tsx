import * as React from 'react';
import {mount} from 'enzyme';
import {IntlProvider} from '../../../src';
import useIntl from '../../../src/components/useIntl';

const FunctionComponent = ({spy}) => {
  const hookReturns = useIntl();
  spy(hookReturns);
  return null;
};

const FC = () => {
  const {formatNumber} = useIntl();
  return formatNumber(10000, {style: 'currency', currency: 'USD'}) as any;
};

describe('useIntl() hook', () => {
  beforeEach(() => {
    console.error = jest.fn();
  });

  it('throws when <IntlProvider> is missing from ancestry', () => {
    expect(() => mount(<FunctionComponent />)).toThrow(
      '[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.'
    );
  });

  it('hooks onto the intl context', () => {
    const spy = jest.fn();
    const rendered = mount(
      <IntlProvider locale="en">
        <FunctionComponent spy={spy} />
      </IntlProvider>
    );
    const intl = rendered.state('intl');
    expect(spy).toHaveBeenCalledWith(intl);
  });

  it('should work when switching locale on provider', () => {
    const rendered = mount(
      <IntlProvider locale="en">
        <FC />
      </IntlProvider>
    );
    expect(rendered).toMatchSnapshot();
    rendered.setProps({
      locale: 'es',
    });
    expect(rendered).toMatchSnapshot();
    rendered.setProps({
      locale: 'en',
    });
    expect(rendered).toMatchSnapshot();
  });
});
