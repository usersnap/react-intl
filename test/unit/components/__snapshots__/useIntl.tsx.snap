// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useIntl() hook should work when switching locale on provider 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <FC>
    $10,000.00
  </FC>
</IntlProvider>
`;

exports[`useIntl() hook should work when switching locale on provider 2`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="es"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <FC>
    10.000,00Â US$
  </FC>
</IntlProvider>
`;

exports[`useIntl() hook should work when switching locale on provider 3`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <FC>
    $10,000.00
  </FC>
</IntlProvider>
`;
